### ISSUE TITLE
.6352WARNING:tensorflow:Your input ran out of data; interrupting training. 

### ISSUE THREAD

**Issue by Asuna88** [zxq2233/n2n-watermark-remove#10]
博主，您好，训练时遇到了这个问题，请你看下，谢谢！

“

.6352WARNING:tensorflow:Your input ran out of data; interrupting training. Make sure that your dataset or generator can generate at least `steps_per_epoch * epochs` batches (in this case, 100000 batches). You may need to use the repeat() function when building your dataset

”

我在代码里面没有找到类似 data.repeat()这样可以添加的语句，请指点，谢谢！

**Comment by panghu955** [comment]
我用tf2也遇到了，降级tf到1.14就行了

**Comment by flashfireXia** [comment]
> 

你要保障你epochs* steps_per_epoch小于等于你的训练样本(train里面图片的数量)。train.py的42行显示默认的steps_per_epoch是1000，38行默认100的nb_epochs，所以train里的图片是至少是1000*100=100000个。在启动训练的时候：python train.py --image_dir dataset/train --test_dir dataset/test --image_size 128 --batch_size 8 --nb_epochs 50 --lr 0.001 --source_noise_model text,0,50 --target_noise_model text,0,50 --val_noise_model text,25,25 --loss mae --output_path text_noise 你可以通过限制--nb_epochs或者steps_per_epoch，使得二者乘积小于你的train的图片数，但是不建议你这么做，因为很影响效果。我建议你直接下载coco2017的那个train，实在懒得下，你直接把你train里面的数据复制几百次。作者是为了快，用val里面那几千个图片作为训练数据。
